### ISSUE TITLE
[WKT2/Proj6] : East / West longitude systems

### ISSUE THREAD

**Issue by thareUSGS** [USGS-Astrogeology/GDAL_scripts#39]
Issue: The planetary community has the need to support both positive East and positive West longitude directions. I do not think the use of positive West longitudes are available within the Well Know Text (WKT) 2 standard.

Question: Can a WKT2 definition support East and West longitudes? 

Note that **technically**, map projection implementations all assume East longitudes for their equations. Thus perhaps the use of East/West with WKT2 is beyond the capability of the standard. I will note that PDS4 has implemented a primary and secondary target definition to allow for both positive East and positive West bounding coordinates to meet historical uses.

Background: In 1970, the IAU  general   assembly   approved   the   use   of   “planetographic”   coordinates   for   extraterrestrial mapping, consisting of geographic latitudes (i.e., measured normal to the ellipsoid, equivalent to geographic or geodetic latitude on the target body) and the **direction of positive  longitude  chosen  such  that  the  longitudes  seen  by  an  outside  observer  increase  with time.**  This longitude convention follows traditional astronomical usage, and results in  positive  West  longitude  for  Mars  and  many  other  bodies (unfortunately opposite of the positive East definition used for more Earth-based applications).

**Comment by thareUSGS** [comment]
To be clear, I have always understood that positive West coordinates are only allowed in geographic latitude systems. Any body defined using ocentric latitudes must use East coordinates.

I don't have a clear path forward for this issue.

While I would like to support positive West coordinates longitudes, but because of how map projections are technically used (see above), it perhaps might be too much to support within the WKT2 standard. Perhaps this can be supported only in applications like ISIS3, VICAR, PDS4 labels, ect. FOr map projected data sets, using positive West coordinates does not impact the registration of a data in meters (upper left pixel location in meters), it is impacted on how one defines the center longitude.

Perhaps using East coordinates in a WKT2 definition but including a WKT2 **comment** that this body should **report** longitudes in West coordinates is enough?

**Comment by desruisseaux** [comment]
WKT 2 supports positive west longitudes, it should not be a problem. For geographic CRS (note the `WEST` keyword in last `Axis` element):

```
GeogCRS["...",
  Datum[...],
  CS[ellipsoidal,2],
    Axis[“latitude”, north],
    Axis[“longitude”, west],
    ...]
```

For projected CRS:

```
ProjCRS["...",
  BaseGeogCRS["...",
    Datum[...]],
  Conversion[“...”],
  CS[Cartesian,2],
    Axis["(Y)", north],
    Axis["(X)", west]],
    ...]
```

It is up to software implementations to handle themselves the sign reversal before to apply the map projection formulas. Apache SIS does that automatically; I think PROJ 6 can do that too. So I would said that there is no issue there; just declare "west" direction in "longitude" or "X" axis.

**Comment by thareUSGS** [comment]
thanks for clarifying. We will set "west" for applicable uses and assume the application can deal with it. I will close this.

**Comment by rouault** [comment]
Yes PROJ >= 6 can deal with 'west' direction in CS[] definition and apply the sign inversion when doing the maths.

**Comment by desruisseaux** [comment]
I realized later that [OGC specification §7.5.4](http://docs.opengeospatial.org/is/18-010r7/18-010r7.html#48) said: _"For geographic CRSs having an ellipsoidal 2-D coordinate system, (…snip…) Axis direction shall be ‘north’ and ‘east’ respectively."_ It seems to exclude the possibility of other directions like "west". However since both PROJ 6 and Apache SIS support the "west" direction, I suggest to keep considering this issue as resolved. But we could open a wiki page somewhere for writing down the issues to bring to OGC after we finished collecting them.
